### PR TITLE
Migration of base_report_to_printer

### PR DESCRIPTION
**Pull request by guewen**
# Migration of base_report_to_printer
- Migration to the new API
- Compatible with the new reports (`report` module, but only the `pdf` ones, not implemented for `html` reports, I don't think that's useful)
- Compatible with deprecated reports (rml, webkit, ...)
- Removed the `Lock()` local to the process and replaced it by a database lock so the processes share the same lock and last update timestamp
- Refreshment of the views

### CODE REVIEW

File: base_report_to_printer/users.py [hunk 1]
```diff
@@ -22,27 +21,23 @@
...
+                in _available_action_types(self)
+                if code != 'user_default']
 
-    _columns = {
-        'printing_action': fields.selection(_user_available_action_types, 'Printing Action'),
-        'printing_printer_id': fields.many2one('printing.printer', 'Default Printer'),
-        }
-
-# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
+    printing_action = fields.Selection(_user_available_action_types)
```
Comment (yvaucher) [inline base_report_to_printer/users.py:41]: Field string 'Printing Action' was removed intentionally ?

Comment (guewen) [inline base_report_to_printer/users.py:41]: Yes.
As advised in https://www.odoo.com/documentation/8.0/reference/orm.html in **Porting from the old API**:

> if a field’s string is the titlecased version of the field name:
> `name = fields.Char(string="Name")`
> it is useless and should be removed

Comment (yvaucher) [inline base_report_to_printer/users.py:41]: ok

**Comment by yvaucher** [comment]
Great cleaning and thanks for the job done here :+1:

**Comment by guewen** [comment]
I will make a few changes (extract a method at least) to reduce the code in printer_tray.

**Comment by guewen** [comment]
Done.

**Comment by yvaucher** [comment]
There is an issue on install: Programming error: relation "printing_printer_polling" does not exist

**Comment by guewen** [comment]
WTF

@yvaucher I fixed it

**Comment by yvaucher** [comment]
:+1:

**Comment by guewen** [comment]
but my last commit is bad

**Comment by guewen** [comment]
Pushed the fix.

**Comment by pedrobaeza** [comment]
:+1: 

Thanks for the conversion to new API.

**Pull request merged**